include/pybind11/pybind11.h

```diff
@@ -2535,7 +2535,7 @@ class exception : public object {
     }
 
     // Sets the current python exception to this exception object with the given message
-    void operator()(const char *message) { PyErr_SetString(m_ptr, message); }
+    void operator()(const char *message) { detail::raise_err(m_ptr, message); }
```

Comment:
Is this tested in this PR? This would allow for nested python exception that raised via raise from (ie. if the Python err indicator is already set) etc,

Reply:
Yes, this is essentially tested by the existing unit tests for `register_exception()` and also by the new `test_throw_custom_nested_exception()`. It just makes the custom exception translator "nested-aware" by replacing

```
PyErr_SetString(m_ptr, message);
```

with

```
if (PyErr_Occurred()) {
  raise_from(m_ptr, message);
} else {
  PyErr_SetString(m_ptr, message);
}
```